src/components/smartSearch/filters/Random/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
+import { FormEvent, useEffect, useState } from 'react';
+
+import { getQuantityWithConfig } from 'components/smartSearch/utils';
+import StyledNumberInput from '../../inputs/StyledNumberInput';
+import StyledSelect from '../../inputs/StyledSelect';
+import useSmartSearchFilter from 'hooks/useSmartSearchFilter';
+import { NewSmartSearchFilter , OPERATION, QUANTITY, RandomFilterConfig, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
+
+const shouldGenerateSeed = (
+    filter: ZetkinSmartSearchFilter<RandomFilterConfig>,
+    initialFilter: SmartSearchFilterWithId<RandomFilterConfig>) => {
+    const isInt = (num: number) => num % 1 === 0;
+    const currentSize = filter.config.size;
+    const initialSize = initialFilter.config.size;
+    return (
+        (isInt(currentSize) !== isInt(initialSize)) ||
+        (isInt(currentSize) && isInt(initialSize) && currentSize > initialSize) ||
+        (!isInt(currentSize) && !isInt(initialSize) && currentSize > initialSize)
```

Comment:
Correct me if I'm wrong, but if the first condition is true (`isInt(currentSize) != isInt(initialSize)`) the other two won't be evaluated, and if it's false, that means that `currentSize` and `initialSize` are the same type (int/not int). So it should not be necessary to do the `isInt()` checks in conditions two and three. I.e. the following should be identical in function:

```tsx
return (
  (isInt(currentSize) !== isInt(initialSize)) ||
  (currentSize > initialSize)
);
```

Reply:
Also it might be nice with a little comment here explaining the logic, which is easy for me to understand because I suggested it, but probably less so for other readers. 😊